src/main/redux/sagas/auth.ts

```diff
@@ -558,16 +568,28 @@ interface IOPDSAuthDocParsed {
 
 }
 function opdsAuthDocConverter(doc: OPDSAuthenticationDoc, baseUrl: string): IOPDSAuthDocParsed | undefined {
-
     if (!doc || !(doc instanceof OPDSAuthenticationDoc)) {
+        // Todo: Return typed error so user-friendly message can be displayed
+        debug("OPDS Authentication Document is not an instance of the model");
+        return undefined;
+    }
 
-        debug("opds authentication doc is not an instance of the model");
+    // https://drafts.opds.io/authentication-for-opds-1.0#231-core-properties
+    if (!doc.Id || doc.Id === "") {
```

Comment:
Authentication ID must be set https://drafts.opds.io/authentication-for-opds-1.0#233-example:~:text=The%20Authentication%20Document%20must%20contain%20the%20following%20keys%3A

but the opds parser is mostly not opinionated and allow to parse badly formatted opds format by making assumptions. Currently the Id is not checked, I wonder if this will invalidate misconfigured opds servers unnecessarily.

@danielweck What is your opinion ?

Reply:
Updated PR to ensure backwards compatibility with already misconfigured OPDS Authentication Providers.